.gitignore

```diff
@@ -146,3 +146,5 @@ personal_scripts/
 
 # Logs
 debug.log
+
+.lock
```

Comment:
Done. Thank you. No, It-s only a thing of mine: in general I don't like centralized versions management. Someone will end up committing his tests by mistake. But never mind, It-s not a real problem, just a preference.

Reply:
No problemo, that's why we do PR reviews 😁 I used to do the same but in the end, it's always recommended to commit lockfiles, it's what they are designed for. This counts for any lockfile, such as Node's package lockfile, PHP's composer lockfile, ...

The main problem it solves is the "It works on my machine" problem.

For example, on a CI deploy, you'd never ever run `npm install` but always `npm ci` which will only install packages from lockfiles so that whatever is working for the devs will end up running on the server

They're no 100% foolproof, but they  get the job done usually, another great explanation about them can be found here: https://blog.shalvah.me/posts/understanding-lockfiles